extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -119,4 +124,21 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
   {
     inspector.visit("selector", selector);
   }
+
+  @Override
+  public void relocate(int oldPosition, int newPosition, ByteBuffer newBuffer, ByteBuffer oldBuffer)
+  {
+    createNewUnion(newBuffer, newPosition, true);
```

Comment:
we need to get rid of old Union object and possibly oldBuffer from the cache (if this was the last relocate call)